File: java/spring-hibernate/src/main/java/com/sprint/sms/api/domain/RolePath.java
package com.sprint.sms.api.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.ideahut.common2.annotation.IdhFormatter;
import com.ideahut.shared2.domain.base.StringIdEntryDomain;

@Entity
@Table(name = "t_role_path", uniqueConstraints = {@UniqueConstraint(columnNames = {"f_role", "path"})})
@IdhFormatter
public class RolePath extends StringIdEntryDomain {
	/**
	 * 
	 */
	private static final long serialVersionUID = -459517068833168887L;
	
	private Role role;
	
	private String path; // Request Mapping	

	@ManyToOne(targetEntity = Role.class, fetch = FetchType.EAGER)
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JoinColumn(name = "f_role", nullable = false)
	@IdhFormatter(field = "id")
	public Role getRole() {
		return role;
	}

	public void setRole(Role role) {
		this.role = role;
	}
	
	@Column(name = "path", nullable = false)
	@IdhFormatter
	public String getPath() {
		return path;
	}

	public void setPath(String path) {
		this.path = path;
	}
	
}

File: php/slim-doctrine/app/src/com/sprint/sms/api/domain/RolePath.php
<?php
namespace App\com\sprint\sms\api\domain;

use Doctrine\ORM\Mapping as ORM;

use \App\com\sprint\sms\api\base\domain\BaseEntry;
use \App\com\sprint\sms\api\support\FormatSupport;

use \App\com\sprint\sms\api\domain\Role;

/**
 * @ORM\Entity
 * @ORM\Table(name = "t_role_path", uniqueConstraints = {@ORM\UniqueConstraint(name = "role_path_idx", columns = {"f_role", "path"})}) 
 */
class RolePath extends BaseEntry implements FormatSupport
{

	/**
     * @ORM\Column(name="ID_", type = "string")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy = "UUID")
     */
	private $id;
	
	/**
	 * @ORM\ManyToOne(targetEntity = "Role", fetch = "EAGER")
	 * @ORM\JoinColumn(name = "f_role", referencedColumnName = "ID_", nullable = false, onDelete="CASCADE")
	 */ 
    private $role;
	
	/**
     * @ORM\Column(name = "path", type = "string", nullable = false)
     */
    private $path;
	
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}	
	
	public function setRole(Role $role)
    {
        $this->role = $role;
    }
	
    public function getRole()
    {
        return $this->role;
    }
	
	public function setPath($path) {
		$this->path = $path;
	}
	
	public function getPath() {
		return $this->path;
	}
	
	public function toFormatObject() {
		$o = parent::toFormatObject();
		$o["id"] = $this->id;
		$o["role"] = isset($this->role) ? array("id" => $this->role->getId()) : null;
		$o["path"] = $this->path;
		return $o;
	}
	
}